src/components/GoodsList/GoodsList.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+export class GoodsList extends React.Component {
+  state = {
+    isVisible: false,
+    isReversed: false,
+    sortBy: '',
+    goods: this.props.goods,
+  };
+
+  showList = () => {
+    this.setState(state => ({
+      isVisible: !state.isVisible,
+    }));
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  }
+
+  sortByAlphabet = () => {
+    this.setState({
+      sortBy: 'alphabet',
+    });
+  }
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+  }
+
+  render() {
+    const { goods, isVisible, isReversed, sortBy } = this.state;
+    const sortedGoods = goods.filter(item => item);
+
+    sortedGoods.sort((a, b) => {
+      switch (sortBy) {
+        case 'alphabet':
+          return a.localeCompare(b);
+        case 'length':
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    return (
+      <div className="App">
+        <h1>Goods</h1>
+        {!isVisible
+          ? (
+            <button
+              type="button"
```

Comment:
This button and `isVisible` property should be on the upper level (in the `App`) and according to `isVisible` you should render the list:
`{isVisible && <GoodsList goods={goodsFromServer} />)`